### ISSUE TITLE
`space-before-function-paren` do false positive at async functions.

### ISSUE THREAD

**Issue by mysticatea** [babel/babel-eslint#207]
Today, I got a `space-before-function-paren` warning in the following code.

```js
import runTask from "./run-task";

export default async function runTasksInSequencial(tasks, stdin, stdout, stderr) {
    for (const task of tasks) {
        const result = await runTask(task, stdin, stdout, stderr);
        if (result.code) {
            throw new Error(`${result.task}: None-Zero Exit(${result.code});`);
        }
    }
}
```

```
3:30  error  Unexpected space before function parentheses  space-before-function-paren
```

https://travis-ci.org/mysticatea/npm-run-all/jobs/91133881

- eslint v1.9.0
- babel-eslint v4.1.5

I had not got this warning until 3 days ago, probably I had been using the same version.
([this](https://github.com/babel/acorn-to-esprima/commit/482350ca852f2b5cd5007212c535424ad623c118) might be related)

**Comment by hzoo** [comment]
=/ That's what I was mentioning with https://github.com/eslint/eslint/issues/4415#issuecomment-156594730. What do you think we can do since its obvious that these 2 rules don't work together

**Comment by mysticatea** [comment]
Hmm, OK.
This problem might be fixed when my code-path-analysis branch is merged because I modified that rule to improve performance in that branch.

Before: https://github.com/eslint/eslint/blob/master/lib/rules/space-before-function-paren.js#L69
After: https://github.com/mysticatea/eslint/blob/code-path-analysis/new/lib/rules/space-before-function-paren.js#L67

**Comment by alexeyraspopov** [comment]
This issue also happens to me when I'm using `babel-eslint@4.1.5`. Everything is ok with `v4.1.3`.

**Comment by alexeyraspopov** [comment]
Some just made a horrible mistake https://github.com/babel/acorn-to-esprima/issues/4

**Comment by hzoo** [comment]
It was a bug fix for https://github.com/eslint/eslint/issues/4415 that resulted in other problems. Made 1.0.7 to revert.
